test/jani_generator/_test_data/string_comparison/string_receive.scxml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scxml
+    name="string_receive"
+    version="1.0"
+    initial="init"
+    model_src=""
+    xmlns="http://www.w3.org/2005/07/scxml">
+
+    <datamodel>
+        <data id="string_store" expr="''" type="string" />
+        <data id="comparison" expr="false" type="bool" />
+    </datamodel>
+
+    <state id="init">
+        <onentry>
+            <send event="string_eval">
+                <param name="comparison" expr="string_store == 'some information'" />
+            </send>
+        </onentry>
+        <transition target="init" event="string_event" cond="_event.data.string == 'more information'">
+            <assign location="string_store" expr="_event.data.string" />
+            <send event="comp_result">
+                <param name="res" expr="1" />
+            </send>
+        </transition>
+        <transition target="init" event="string_event">
+            <assign location="string_store" expr="_event.data.string" />
+            <assign location="comparison" expr="string_store == 'some information'" />
+            <if cond="comparison">
+                <send event="comp_result">
+                    <param name="res" expr="2" />
+                </send>
+                <else/>
+                <send event="comp_result">
+                    <param name="res" expr="-1" />
+                </send>
+            </if>
+        </transition>
```

Comment:
Does scxml work in the way, that the second transition is only used if the condition of the first is false?

Reply:
In theinterpretation, we automatically generate the negation of the condition of the previous transition(s), and assign it to the current one.